### PR TITLE
[FLINK-27334] Support auto generate the doc for the `KubernetesOperatorConfigOptions`

### PR DESCRIPTION
**Pull request by SteNicholas**
The document of the `KubernetesOperatorConfigOptions` should be automatically generated.

**The brief change log**

- Adds the `flink-kubernetes-docs` module and introduces the `ConfigOptionsDocGenerator` to generate the document of the `KubernetesOperatorConfigOptions`.

### CODE REVIEW

**Review by wangyang0918 — COMMENTED** [review]
Thanks for creating this PR. I left some comments and please have a look.

File: Dockerfile [hunk 1]
```diff
@@ -23,21 +23,23 @@ WORKDIR /app
...
 ENV WEBHOOK_DIR=flink-kubernetes-webhook
+ENV DOCS_DIR=flink-kubernetes-docs
 
 RUN mkdir $OPERATOR_DIR $WEBHOOK_DIR
 
 COPY pom.xml .
 COPY $SHADED_DIR/pom.xml ./$SHADED_DIR/
 COPY $WEBHOOK_DIR/pom.xml ./$WEBHOOK_DIR/
 COPY $OPERATOR_DIR/pom.xml ./$OPERATOR_DIR/
+COPY $DOCS_DIR/pom.xml ./$DOCS_DIR/
```
[the remark this answers is not in the document]
Comment (SteNicholas) [reply Dockerfile:34]: @wangyang0918, this only adds the pom.xml of the `flink-kubernetes-docs`, which is used in parent pom.xml. I have tried not to copy this file but the maven command throwed the module not existing.

**Comment by SteNicholas** [comment]
@wangyang0918, thanks for your review. I have addressed above comments and verified the configuration html manually. PTAL.
cc @gyfora .

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -80,6 +80,11 @@ jobs:
           cd flink-kubernetes-webhook
           mvn verify -Dit.skip=false
           cd ..
+      - name: Tests in flink-kubernetes-docs
```
Comment (wangyang0918) [inline .github/workflows/ci.yml:83]: We already have `mvn clean install -DskipTests -Pgenerate-docs` in ci.yml to guard the docs are re-generated accordingly. What I mean is to simply add `-Dgenerate-config-docs` here.
Maybe `-Pgenerate-config-docs` looks better.

Comment (gyfora) [reply .github/workflows/ci.yml:83]: Could we simply reuse the existing profile?

Comment (wangyang0918) [reply .github/workflows/ci.yml:83]: It also makes sense to reuse the profile. It seems we do not need to differentiate generating CRD docs and config docs.

**Review by wangyang0918 — APPROVED** [review]
Thanks for addressing all the comments. Only have last one minor comments.

**Comment by wangyang0918** [comment]
@SteNicholas Just to be sure, do you have verified that the CI will fail when the docs are not generated properly?

File: flink-kubernetes-docs/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+This module contains generators that create HTML files directly from Flink Kubernetes Operator's source code.
+
+## Configuration documentation
+
+The `ConfigOptionsDocGenerator` can be used to generate a reference of `ConfigOptions`. By default, a separate file is generated for each `*Options` class found in `org.apache.flink.kubernetes.operator.docs.configuration`. The `@ConfigGroups` annotation can be used to generate multiple files from a single class.
+
+To integrate an `*Options` class from another package, add another module-package argument pair to `ConfigOptionsDocGenerator#LOCATIONS`.
+
+The files can be generated by running `mvn package -Dgenerate-docs -pl flink-kubernetes-docs -nsu -DskipTests`, and can be integrated into the documentation using `{{ include generated/<file-name> >}}`.
```
Comment (wangyang0918) [inline flink-kubernetes-docs/README.md:30]: We need to update the docs here.

**Review by wangyang0918 — CHANGES_REQUESTED** [review]

**Comment by SteNicholas** [comment]
> @SteNicholas Just to be sure, do you have verified that the CI will fail when the docs are not generated properly?

@wangyang0918 , I'm sure that the CI will fail when the docs are not generated properly in updates of the pull request.

File: flink-kubernetes-docs/README.md (hunk 3 above)
Comment (SteNicholas) [reply flink-kubernetes-docs/README.md:30]: @wangyang0918, this has already been updated for the profile `generate-docs`. I didn't get where we need to update in this doc.

**Review by wangyang0918 — APPROVED** [review]
+1 for merging.

**Pull request merged**